### DISCUSSION TITLE
Would like to customize a different ohmic heater

### DISCUSSION THREAD

**Discussion by Dudic0** [Ralim/IronOS#1147] [Ideas]
Hi,
I Love the work done so far, thanks!
I would like to customize the system to a specific heater, is there any easy way to change heater values like resistance and TCR values? I have a heater with 12ohm in 0c and a TCR of 3850ppm/k.

Thanks and sorry if this is the wrong place for this type of request.

David

**Comment by Ralim** [comment]
Hello,

Changing tip resistance is easy, changing thermal response isn't as simple as you need to recreate the lookup table https://github.com/Ralim/IronOS/blob/f13b7be3d9b0eb5eaa8e95b31b7687d784e19e2f/source/Core/BSP/Pine64/ThermoModel.cpp#L12

To use the new uV<-> temperature curve

**Reply by Dudic0** [reply]
Thanks! so i need to recreate this table using a voltmeter and a reliable heating source.. 

Could you please point me to where do i enter the heater resistance value in the code? Does the parameter represent a certain temperature reference like 25c or 0c and is there also a way to change the thermocouple type and enter its coefficient? I would like to experiment with different TC than in the stock.

Thanks again!
David.

**Reply by Ralim** [reply]
Hia,
So it depends on what device you are modifying as to what you will need to change.
You need to change `TIP_RESISTANCE`. You can find this in `source/Core/BSP/<brand>/configuration.h`. Generally near the bottom.
This is the ohmic resistance of the tip around normal operating temp (its approx, doesnt need to be exact. Not _critical_).
its stored in 100milliohm increments, so 12 ohms would be `120`.

For the Thermocouple, there isnt really a "type" but instead a lookup table for `uV` -> `Temp`.
If you look in `source/Core/BCP/<brand>/ThermoModel.cpp` there is the table for these lookups that you would need to modify (the one suiting your device). To estimate a linear function for testing you could probably get away with just putting in end values and it will linearly interpolate those for you.

**Reply by Dudic0** [reply]
Thank you. I found all references.
Could you share, what would like be the approach to modify  "TIP_THERMAL_MASS         65 // TODO, needs refinement"
As my element has a smaller heating mass..

Thanks again, you are vey helpfull.

**Reply by Ralim** [reply]
Try without setting it first, then if your are overshooting on temp reduce it. If you are underdamped (not hitting 100% power steering great up) increase it.

You can be quite wrong and it will still work just be a little unstable
